sdv/single_table/base.py

```diff
@@ -276,3 +294,648 @@ def update_transformers(self, column_name_to_transformer):
         self._validate_transformers(column_name_to_transformer)
         self._warn_for_update_transformers(column_name_to_transformer)
         self._data_processor.update_transformers(column_name_to_transformer)
+
+    def _set_random_state(self, random_state):
+        """Set the random state of the model's random number generator.
+
+        Args:
+            random_state (int, tuple[np.random.RandomState, torch.Generator], or None):
+                Seed or tuple of random states to use.
+        """
+        self._model.set_random_state(random_state)
+
+    @staticmethod
+    def _filter_conditions(sampled, conditions, float_rtol):
+        """Filter the sampled rows that match the conditions.
+
+        If condition columns are float values, consider a match anything that
+        is closer than the given ``float_rtol`` and then make the value exact.
+
+        Args:
+            sampled (pandas.DataFrame):
+                The sampled rows, reverse transformed.
+            conditions (dict):
+                The dictionary of conditioning values.
+            float_rtol (float):
+                Maximum tolerance when considering a float match.
+
+        Returns:
+            pandas.DataFrame:
+                Rows from the sampled data that match the conditions.
+        """
+        for column, value in conditions.items():
+            column_values = sampled[column]
+            if column_values.dtype.kind == 'f':
+                distance = value * float_rtol
+                sampled = sampled[np.abs(column_values - value) <= distance]
```

Comment:
Not sure if this works with nans (if it's possible for nans to reach this stage).

Reply:
@pvk-developer Did you get a chance to check this?

Reply:
This returns empty  `pd.DataFrame` if by any chance this occurs or reaches this point, which I couldn't make reach.

Reply:
This is how the current implementation is as well. I think it's fine for now. We should keep an eye on any bugs for this scenario